src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1222,146 +1221,30 @@ private IArrayInitializerOperation CreateBoundArrayInitializationOperation(Bound
-                    spanConstructorParameter?.Type,
-                    isImplicit: true);
-                IArgumentOperation constructorArgument = new ArgumentOperation(
-                    ArgumentKind.Explicit,
-                    spanConstructorParameter,
-                    array,
-                    OperationFactory.IdentityConversion,
-                    OperationFactory.IdentityConversion,
-                    _semanticModel,
-                    syntax,
-                    isImplicit: true);
-                return new ObjectCreationOperation(
-                    spanConstructor,
-                    initializer: null,
-                    ImmutableArray.Create(constructorArgument),
-                    _semanticModel,
-                    syntax,
-                    collectionType,
-                    constantValue: null,
-                    isImplicit);
-            }
-        }
+            return new NoneOperation(elements, _semanticModel, syntax, type: type, constantValue: null, isImplicit);
 
-        private IOperation CreateBoundCollectionInitializerCollectionLiteralExpression(BoundCollectionLiteralExpression boundCollectionLiteralExpression)
-        {
-            SyntaxNode syntax = boundCollectionLiteralExpression.Syntax;
-            ITypeSymbol? collectionType = boundCollectionLiteralExpression.GetPublicTypeSymbol();
-            bool isImplicit = boundCollectionLiteralExpression.WasCompilerGenerated;
-            var initializer = new ObjectOrCollectionInitializerOperation(
-                CreateFromArray<BoundExpression, IOperation>(boundCollectionLiteralExpression.Elements),
-                _semanticModel,
-                syntax,
-                collectionType,
-                isImplicit: true);
-            if (collectionType is ITypeParameterSymbol)
-            {
-                return new TypeParameterObjectCreationOperation(initializer, _semanticModel, syntax, collectionType, isImplicit: isImplicit);
-            }
-            else if (boundCollectionLiteralExpression.CollectionCreation is BoundObjectCreationExpression collectionCreation)
-            {
-                return new ObjectCreationOperation(
-                    collectionCreation.Constructor.GetPublicSymbol(),
-                    initializer,
-                    DeriveArguments(collectionCreation),
-                    _semanticModel,
-                    syntax,
-                    collectionType,
-                    constantValue: null,
-                    isImplicit: isImplicit);
-            }
-            else
+            IOperation createChild(BoundExpression expression)
             {
-                return new InvalidOperation(ImmutableArray.Create<IOperation>(initializer), _semanticModel, syntax, collectionType, constantValue: null, isImplicit);
+                var element = expression switch
+                {
+                    BoundCollectionElementInitializer initializer => initializer.Arguments.Single(),
```

Comment:
This is fine for now, but for the final shape we are almost certainly going to want to reflect more info from the actual collection initializer. We know already that users of our API want info about collection initialization (which is why we have the deprecated `ICollectionElementInitializerOperation` type that we abanded when users wanted to get more info), so there is prior art for exposing more info here.